tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,180 @@
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/26",  None),    # Smallest legal pool
+        (True,  "CALICO_IPV6POOL_CIDR", "fd00::/122",   None),    # Smallest legal pool
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/123",   None),    # Too small
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/128",   None),    # Too small, but legal
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/129",   None),    # Impossible CIDR
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "cross-subnet"),  # Typ. non-def pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "always"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "off"),  # Typical pool, explicitly no IPIP
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "always"),  # IPv6 - IPIP not permitted
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "cross-subnet"),  # IPv6 - IPIP not permitted
+        (True, "CALICO_IPV6POOL_CIDR", "fd00::/122", "off"),      # IPv6, IPIP explicitly off
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/122", "junk"),     # Invalid IPIP value
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "reboot"),  # Invalid IPIP value
+        (True,  "CALICO_IPV4POOL_CIDR", "0.0.0.0/0",    None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "::/0",         None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "fd80::0:0/120", None),   # Valid, but non-canonical form
+        (False, "CALICO_IPV6POOL_CIDR", "1.2.3.4/24",    None),   # Wrong type
+        (False, "CALICO_IPV4POOL_CIDR", "fd00::/24",     None),   # Wrong type
+        (True,  "CALICO_IPV6POOL_CIDR", "::0:a:b:c:d:e:0/120", None),  # Valid, non-canonical form
+        (False, "CALICO_IPV4POOL_CIDR", "1.2/16",       None),    # Valid, unusual form
+    ])
+    def test_default_pools(self, success_expected, param, value, ipip):
+        """
+        Test that the various options for default pools work correctly
+        """
+        # Get command line for starting docker
+        output = self.host.calicoctl("node run --dryrun")
+        base_command = output.split('\n')[-4].rstrip()
+
+        # Modify command line to add the options we want to test
+        # Command line will attempt to use quay.io/calico/node:latest - we want calico/node:latest
+        base_command = base_command.replace("quay.io/", "")
+        env_index = base_command.find("-e")
+        command = base_command[:env_index] + "-e %s=%s " % (param, value) + base_command[env_index:]
+        if ipip is not None:
+            command = command[:env_index] + \
+                      "-e CALICO_IPV4POOL_IPIP=%s " % ipip + \
+                      command[env_index:]
+
+        # Start calico-docker
+        self.host.execute(command)
+
+        def check():
+            assert expected_string in self.host.execute("docker logs calico-node"), \
```

Comment:
Not keen on this using a "global" my preference would be to move this to its own method:
```
def assert_calico_node_log_contains(expected_msg):
    def check():
        assert ...
    
    retry_until_success(check, ...)
```
Then it can be shared with the other test too.